android/src/app/organicmaps/widget/placepage/PlacePageBookmarkFragment.java

```diff
@@ -0,0 +1,178 @@
+    settings.setJavaScriptEnabled(false);
+    settings.setDefaultTextEncodingName("UTF-8");
+    final LinearLayout linearLayout = mFrame.findViewById(R.id.place_page_bookmark_layout);
+    // Add the webview in last position
+    linearLayout.addView(mWvBookmarkNote, linearLayout.getChildCount() - 1);
+  }
+
+  @Nullable
+  private MapObject getMapObject()
+  {
+    if (viewModel != null)
+      return viewModel.getMapObject().getValue();
+    return null;
+  }
+
+  @Override
+  public void onDestroy()
+  {
+    super.onDestroy();
+    viewModel.getMapObject().removeObserver(this);
+  }
+
+  private void updateBookmarkDetails(@NonNull MapObject mapObject)
+  {
+    if (mapObject.getMapObjectType() != MapObject.BOOKMARK)
+      return;
+
+    final String notes = ((Bookmark) mapObject).getBookmarkDescription();
+    if (TextUtils.isEmpty(notes))
+    {
+      UiUtils.hide(mTvBookmarkNote);
+      if (mWvBookmarkNote != null)
+        UiUtils.hide(mWvBookmarkNote);
+      return;
+    }
+
+    if (StringUtils.nativeIsHtml(notes))
+    {
+      // According to loadData documentation, HTML should be either base64 or percent encoded.
+      // Default UTF-8 encoding for all content is set above in WebSettings.
+      initWebView();
+      if (mWvBookmarkNote != null)
+      {
+        final String b64encoded = Base64.encodeToString(notes.getBytes(), Base64.DEFAULT);
+        mWvBookmarkNote.loadData(b64encoded, Utils.TEXT_HTML, "base64");
+        UiUtils.show(mWvBookmarkNote);
+        UiUtils.hide(mTvBookmarkNote);
+      }
+    }
+    else
+    {
+      mTvBookmarkNote.setText(notes);
+      Linkify.addLinks(mTvBookmarkNote, Linkify.ALL);
+      UiUtils.show(mTvBookmarkNote);
+      if (mWvBookmarkNote != null)
+        UiUtils.hide(mWvBookmarkNote);
+    }
+  }
+
+  @Override
+  public void onClick(View v)
+  {
+    final MapObject mapObject = getMapObject();
+    if (mapObject == null)
+    {
+      Logger.e(TAG, "A bookmark cannot be edited, mMapObject is null!");
+      return;
+    }
+    Bookmark bookmark = (Bookmark) mapObject;
+    EditBookmarkFragment.editBookmark(bookmark.getCategoryId(),
+                                      bookmark.getBookmarkId(),
+                                      requireActivity(),
+                                      requireActivity().getSupportFragmentManager(),
+                                      PlacePageBookmarkFragment.this);
+  }
+
+  @Override
+  public boolean onLongClick(View v)
+  {
+    final String notes = mTvBookmarkNote.getText().toString();
+
+    final Context ctx = requireContext();
+    Utils.copyTextToClipboard(ctx, notes);
+    Utils.showSnackbarAbove(mFrame,
+                            mFrame.getRootView().findViewById(R.id.pp_buttons_layout),
+                            ctx.getString(R.string.copied_to_clipboard, notes));
+    return true;
+  }
+
+  @Override
+  public void onChanged(MapObject mapObject)
+  {
+    if (mapObject != null)
```

Comment:
I'll look more into this, but when the place page is closed, the mapobject is set to null. The fragment should be destroyed but to make sure I added this check. I can try to remove it and see if it works.

Reply:
When the place page is closed, the mapObject is set to null. The controller detects when this value becomes null and destroys the fragments. But this means the fragments themselves could receive the event that the mapObject became null.

One way to fix this would be to destroy the fragments and then set the value to null, or never set it to null.

Reply:
Is it possible to unsubscribe first, before setting a null map object?

If it's too complex now, then please add your comment above to the code.